### DISCUSSION TITLE
Keycloak integration with Sunbird lern module

### DISCUSSION THREAD

**Discussion by pritirc** [Sunbird-Lern/Community#57] [Installation]
We are trying to run sunbird lms service in local machine and facing the below issues while registering the user.

1. How to add Casandra Storage Federation Provider to KeyCloak, Currently It is showing only LDAP and Kuberoes.
2. User Information is getting stored in cassandra DB when we create a user, But we are unable to fetch the same User Information from KeyCloak. (Update Password call is failing for keyCloak)

Kindly help us addressing the above issues.

**Comment by Krishnaj20** [comment]
tagging @reshmi-nair @rhwarrier

**Reply by rhwarrier** [reply]
@reshmi-nair  please help resolve these queries.

**Reply by reshmi-nair** [reply]
@pritirc First point is not very clear. Can you explain a bit more?
Regarding second point, User info from cassandra is fetched using sunbird-auth SPI. SPI jar is added into keycloak to interact with UserOrg service. Please refer to [sunbird-auth](https://github.com/Sunbird-Lern/sunbird-auth)
